Framework/DataObjects/src/EventList.cpp

```diff
@@ -3106,6 +3147,56 @@ DateAndTime EventList::getPulseTimeMax() const {
   return tMax;
 }
 
+void EventList::getPulseTimeMinMax(Mantid::Kernel::DateAndTime &tMin,
+                                   Mantid::Kernel::DateAndTime &tMax) const {
+  // set up as the minimum available date time.
+  tMax = DateAndTime::minimum();
+  tMin = DateAndTime::maximum();
+
+  // no events is a soft error
+  if (this->empty())
+    return;
+
+  // when events are ordered by pulse time just need the first/last values
+  if (this->order == PULSETIME_SORT) {
+    switch (eventType) {
+    case TOF:
+      tMin = this->events.begin()->pulseTime();
+      tMax = this->events.rbegin()->pulseTime();
+      return;
+    case WEIGHTED:
+      tMin = this->weightedEvents.begin()->pulseTime();
+      tMax = this->weightedEvents.rbegin()->pulseTime();
+      return;
+    case WEIGHTED_NOTIME:
+      tMin = this->weightedEventsNoTime.begin()->pulseTime();
+      tMax = this->weightedEventsNoTime.rbegin()->pulseTime();
+      return;
+    }
+  }
+
+  // now we are stuck with a linear search
+  size_t numEvents = this->getNumberEvents();
+  DateAndTime temp = tMax; // start with the smallest possible value
+  for (size_t i = 0; i < numEvents; i++) {
```

Comment:
again, the code just copied from the existing one using minimal changes... Minmax may be ok, but comparison operator would be complex and the code becomes heterogeneous. "Minimal perturbations"...